File: Python/Remove Palindromic Subsequences.py
class Solution:
    """
    Time:   O(n)
    Memory: O(n)
    """

    def removePalindromeSub(self, s: str) -> int:
        return 1 + (s != s[::-1])


class Solution:
    """
    Time:   O(n)
    Memory: O(1)
    """

    def removePalindromeSub(self, s: str) -> int:
        for i in range(len(s) // 2):
            if s[i] != s[-i - 1]:
                return 2
        return 1

File: Java/Remove Palindromic Subsequences.java
public class Solution {

    /**
     * Time:   O(n)
     * Memory: O(1)
     */
    public int removePalindromeSub(String s) {
        int n = s.length();

        for (int i = 0; i < n / 2; ++i)
            if (s.charAt(i) != s.charAt(n - 1 - i))
                return 2;

        return 1;
    }
}
